test/EFCore.MySql.FunctionalTests/Scaffolding/MySqlDatabaseModelFactoryTest.cs

```diff
@@ -731,6 +731,41 @@ public void Set_referential_action_for_foreign_key()
 DROP TABLE PrincipalTable;");
         }
 
+        [Fact]
+        public void Ensure_constraints_scaffold_with_case_mismatch()
+        {
+            // Important: the inconsistent casing of the referenced table name is intentional. Do not change.
+            Test(
+                @"
+CREATE TABLE `PrincipalTable` (
+  `Id` INT NOT NULL,
+  PRIMARY KEY (`Id`));
+
+CREATE TABLE `DependentTable` (
+  `Id` INT NOT NULL,
+  `ForeignKeyId` INT NOT NULL,
+  PRIMARY KEY (`Id`),
+  CONSTRAINT `ForeignKey_Id`
+    FOREIGN KEY (`ForeignKeyId`)
+    REFERENCES `principaltable` (`Id`)
+);",
+                Enumerable.Empty<string>(),
+                Enumerable.Empty<string>(),
+                dbModel =>
+                {
+                    var principal = dbModel.Tables.FirstOrDefault(t => t.Name == "PrincipalTable");
+                    var dependent = dbModel.Tables.FirstOrDefault(t => t.Name == "DependentTable");
```

Comment:
The table name checks need to be performed case-*in*sensitive (`OrdinalIgnoreCase`), because they would be e.g. saved as `principaltable` and `dependenttable` with `lower_case_table_names=1`.